src/entry-client.js

```diff
@@ -86,7 +86,8 @@ export default function initializeClient(createApp, clientOpts) {
                 router,
                 store,
             });
-            return opts.middleware(to, from, store)
+            return Promise.resolve()
+                .then(() => opts.middleware(to, from, store))
```

Comment:
Using this pattern exceptions throw in middleware will be caught by the promise chain